crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4063,6 +4058,24 @@ impl<'db> TypeInferenceBuilder<'db> {
         }
     }
 
+    /// Given the slice of a `type[]` annotation, return the type that the annotation represents
+    fn infer_subclass_of_type_expression(&mut self, slice: &ast::Expr) -> Type<'db> {
+        match slice {
+            ast::Expr::Name(name) => {
+                let name_ty = self.infer_name_expression(name);
+                if let Some(class_literal) = name_ty.into_class_literal() {
+                    Type::SubclassOf(class_literal.to_subclass_of_type())
+                } else {
+                    Type::Todo
+                }
+            }
+            _ => {
+                self.infer_type_expression(slice);
+                Type::Todo
+            }
```

Comment:
```suggestion
            // TODO: attributes, unions, subscripts, etc.
            _ => {
                self.infer_type_expression(slice);
                Type::Todo
            }
```

Reply:
I always see a `Type::Todo` as something that we can't miss eventually. But we can add a TODO comment just to be sure.

Reply:
I agree -- I just thought it would be useful to indicate to the reader what exactly is still to be done here 😄

I've wondered about making the `Todo` variant `Type::Todo(&'static str)`, to enforce that we _have_ to explicitly say what is still to be done when using the type... but it might just make things more complicated for little benefit :-)